apollo-router/src/services/router_service.rs

```diff
@@ -337,12 +338,25 @@ impl PluggableRouterServiceBuilder {
                 .await
                 .map_err(ServiceBuildError::QueryPlannerError)?;
         let query_planner_service = ServiceBuilder::new().buffered().service(
-            self.plugins.iter_mut().rev().fold(
-                CachingQueryPlanner::new(bridge_query_planner, plan_cache_limit)
-                    .await
-                    .boxed(),
-                |acc, (_, e)| e.query_planning_service(acc),
-            ),
+            CachingQueryPlanner::new(
+                Buffer::new(
+                    self.plugins
+                        .iter_mut()
+                        .rev()
+                        .fold(bridge_query_planner.boxed(), |acc, (_, e)| {
+                            e.query_planning_service(acc)
+                        }),
+                    DEFAULT_BUFFER_SIZE,
+                ),
+                plan_cache_limit,
+            )
+            .await
+            .boxed(), /*self.plugins.iter_mut().rev().fold(
+                          CachingQueryPlanner::new(bridge_query_planner, plan_cache_limit)
+                              .await
+                              .boxed(),
+                          |acc, (_, e)| e.query_planning_service(acc),
+                      ),*/
```

Comment:
Remove this?

Reply:
fixed in [86b9b21](https://github.com/apollographql/router/pull/1484/commits/86b9b21f8aa8da0a916949cf2fbee8f7422a388d)